pysmt/solvers/yices.py

```diff
@@ -73,18 +131,29 @@ def __init__(self, environment, logic, **options):
                         **options)
 
         self.declarations = set()
-
-        self.yices = yicespy.yices_new_context(None)
-        # MG: TODO: Set options!!!
+        self.yices_config = yicespy.yices_new_config()
+        self.yices = None
+        self.yices_params = None
+        self._create_yices_context()
         self.converter = YicesConverter(environment)
         self.mgr = environment.formula_manager
         self.model = None
         self.failed_pushes = 0
         return
 
+    def _create_yices_context(self):
```

Comment:
Mostly because there is a bit of additional code to handle 1) the logic selection 2) setting of parameters (set_params) 3) freeing of configuration. 

We can also probably just isolate the bit about the logic conversion.